Mailchimp integration

mikefowler opened the pull request:
Adds Mailchimp integration via their official gem for subscribing to the CoderDojoSF announcements list. Also reorganizes some CSS into separate files.

Commit:
Adds Mailchimp integration via their official gem for subscribing to the CoderDojoSF announcements list.

mikefowler commented:
@jonmagic, have a look when you have time.

jonmagic commented on app/controllers/application_controller.rb:
```diff
@@ -1,6 +1,10 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
 
+  before_filter :setup_mcapi
```
I would probably add an initializer instead of using a before filter in the controller. See https://github.com/CoderDojoSF/event-app/tree/master/config/initializers

jonmagic commented on app/controllers/application_controller.rb:
```diff
@@ -43,4 +47,11 @@ def authenticate_user!
+  private 
+
+  # Internal: Creates a new instance of the Mailchimp API
+  def setup_mcapi
```
Same note here, but maybe this is how they recommend in their gem setup instructions?

mikefowler replied:
Yea, guilty. This is line for line how they recommend doing it in their sample project (except they're using the new `before_action` hook.

jonmagic commented on app/controllers/application_controller.rb:
```diff
@@ -43,4 +47,11 @@ def authenticate_user!
+
+  # Internal: Creates a new instance of the Mailchimp API
+  def setup_mcapi
+    @mc = Mailchimp::API.new(ENV["MAILCHIMP_API_KEY"])
```
If we do go this route I think that should be `@mc ||= Mailchimp::API.new(ENV["MAILCHIMP_API_KEY"])`

mikefowler replied:
Agreed.

jonmagic commented:
@mikefowler this is awesome :metal: Cool if I submit a PR to your PR? :grin: I made some notes above, but I'd be happy to actually write the code and submit a PR to yours that we can review and decide which route is best. I'd also like to add a friendly error page instead of rendering text on the error states for subscribing. Lemme know, awesome to see this happening.

jonmagic committed:
This creates an EmailList object that presents a simple API for controllers to use.

I haven’t written tests and I’m sure none of this works, but this is
the direction I’m headed.

mikefowler commented:
@jonmagic one note on the above comment: currently the endpoint I added is only being hit via AJAX, hence the `render :text` returns. The errors are actually displayed inline in the form, if they occur. I'm sure there's a way to have it handle both or do this more elegantly, so suggestions welcome.

jonmagic committed:
Update webmock and associated gems.

jonmagic committed:
Environment variables for test suite.

jonmagic committed:
Test EmailList model.

jonmagic committed:
Update vcr gem.

jonmagic committed:
Test newsletter controller.

jonmagic committed:
Update .env.example and .env.test to not include MANDRILL

Commit:
Merge pull request #23 from CoderDojoSF/refactor-into-model

This creates an EmailList object that presents a simple API to use in controllers.

marcyDel commented:
Hey there Mike,

I am the person who runs the CoderDojo down in Mountain View.  Was wondering if the event app being discussed will have any fields for prerequisites, experience level or link to previous dojo events attended?

Marcy

On Nov 25, 2013, at 10:28 AM, Mike Fowler notifications@github.com wrote:

> Merged #22.
> 
> —
> Reply to this email directly or view it on GitHub.

jonmagic commented:
@marcyDel no plans for that yet, but if you could file a new issue on this project outlining what you're looking for we can discuss it there. Thanks!
